Assets/Scripts/Enemy/FollowingEnemy.cs

```diff
@@ -0,0 +1,68 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+[RequireComponent(typeof(NavMeshAgent))]
+
+public class FollowingEnemy : MonoBehaviour
+{
+    [SerializeField] private Transform[] waypoints;
+
+    private NavMeshAgent agent;
+    private Transform player;
+    private bool isFollowing = false;
+    private int index;
+
+    // Start is called before the first frame update
+    private void Start()
+    {
+        agent = GetComponent<NavMeshAgent>();
+        player = FindObjectOfType<PlayerMovement>().transform;
+        agent.SetDestination(waypoints[0].position);
+    }
+
+    private void Update()
+    {
+        if (NavMesh.SamplePosition(agent.transform.position, out NavMeshHit hit, 1f, NavMesh.AllAreas))
+        {
+            if (NavMesh.GetAreaCost((int)Mathf.Log(hit.mask, 2)) >= 4)
+                agent.speed = 1.5f;
+            else
```

Comment:
https://dev.to/programmerbyda1/why-you-should-always-use-curly-braces-for-single-statement-blocks-in-c-41pg

Reply:
Ok, I will use curly braces in single if statements =)